packages/tools/client-debugger/client-debugger-view/src/components/TelemetryView.tsx

```diff
@@ -326,121 +416,47 @@ export function TelemetryView(): React.ReactElement {
+				}}
+			>
+				<DataGridHeader>
+					<DataGridRow style={{ whiteSpace: "normal" }}>
+						{({ renderHeaderCell }): JSX.Element => (
+							<DataGridHeaderCell>{renderHeaderCell()}</DataGridHeaderCell>
+						)}
+					</DataGridRow>
+				</DataGridHeader>
+				<DataGridBody<Item>>
+					{({ item, rowId }): JSX.Element => (
+						<DataGridRow<Item> key={rowId}>
+							{({ renderCell }): JSX.Element => (
+								<DataGridCell>{renderCell(item)}</DataGridCell>
 							)}
 						</DataGridRow>
-					</DataGridHeader>
-					<DataGridBody<Item>>
-						{({ item, rowId }): JSX.Element => (
-							<DataGridRow<Item> key={rowId}>
-								{({ renderCell }): JSX.Element => (
-									<DataGridCell>{renderCell(item)}</DataGridCell>
-								)}
-							</DataGridRow>
-						)}
-					</DataGridBody>
-				</DataGrid>
-			</>
-		) : (
-			<Waiting label={"Waiting for Telemetry events"} />
-		);
-
-	return (
-		<Stack>
-			<StackItem>
-				<_ListLengthSelection
-					currentLimit={maxEventsToDisplay}
-					onChangeSelection={(key): void => setMaxEventsToDisplay(key)}
-				/>
-			</StackItem>
-			<StackItem>{log_view}</StackItem>
-		</Stack>
-	);
-}
-
-/**
- * {@link _ListLengthSelectionProps} input props.
- */
-interface _ListLengthSelectionProps {
-	/**
-	 * The current limit (max number of telemetry events to show).
-	 * @defaultValue {@link DEFAULT_PAGE_SIZE}
-	 */
-	currentLimit: number;
-
-	/**
-	 * Called when the selection changes.
-	 */
-	onChangeSelection(newLimit: number): void;
-}
-
-/**
- * A dropdown menu for selecting how many logs to display on the page.
- */
-function _ListLengthSelection(props: _ListLengthSelectionProps): React.ReactElement {
-	const { currentLimit, onChangeSelection } = props;
-	// const [eventsDisplayed, setEventsDisplayed] = useState(50);
-	const stackTokens: IStackTokens = { childrenGap: 20 };
-
-	// Options formatted for the Fluent Dropdown component
-	const dropdownOptions: { key: number; text: string }[] = [
-		{ key: 50, text: "50" },
-		{ key: 100, text: "100" },
-		{ key: 500, text: "500" },
-		{ key: 1000, text: "1000" },
-	];
-
-	const handleMaxEventChange: DropdownProps["onOptionSelect"] = (event, data) => {
-		onChangeSelection(Number(data.optionText));
-	};
-
-	return (
-		<Stack tokens={stackTokens}>
-			<div className="list-size-options">
-				<h3>Max number of telemetry events to display: </h3>
-				<Dropdown
-					placeholder="Select an option"
-					size="small"
-					style={{ minWidth: "300px", zIndex: "1" }}
-					defaultValue={currentLimit.toString()}
-					// change the number of logs displayed on the page
-					onOptionSelect={handleMaxEventChange}
-				>
-					{dropdownOptions.map((option) => {
-						return (
-							<Option style={{ minWidth: "120px" }} key={option.key}>
-								{option.text}
-							</Option>
-						);
-					})}
-				</Dropdown>
-			</div>
-		</Stack>
+					)}
+				</DataGridBody>
+			</DataGrid>
+		</>
+	) : (
+		<Waiting label={"Waiting for Telemetry events"} />
```

Comment:
The Waiting handling should probably go in the parent component, since that's the one that is communicating with the app. This component shouldn't have to worry about the `undefined` case.